### ISSUE TITLE
Infinite redirect issue

### ISSUE THREAD

**Issue by fifthsegment** [e2guardian/e2guardian#92]
I've experienced this problem with some sites while testing out e2Guardian (e2g default configuration + chrome v46). Here's an example site that causes it: https://www.coursera.org. Whenever I visit the site, I'm unable to get past the chrome error "ERR_TOO_MANY_REDIRECTS". I think this might be caused by sites that issue a 301 http response code on their root domain. Does anyone else experience this too?

**Comment by fifthsegment** [comment]
Correction: My dansguardianf1.conf has mitmcheckcert set to mitmcheckcert = on.

**Comment by fredbcode** [comment]
And with FF or IE ?

**Comment by fifthsegment** [comment]
FF: The page isn't redirecting properly
IE: Waiting for coursera.org (keeps going on)

**Comment by fredbcode** [comment]
> mitmcheckcert set to mitmcheckcert = off. 
So you are using SSLMITM ? Right ?
No problem here (but without SSLMITM) with e2 3.2.0 and https://www.coursera.org
Do you have something in your logs ? 

Can you also check your computer and server time (hour,minutes,year)

**Comment by fifthsegment** [comment]
Yes, I'm using sslmitm for parental controls. Works awesome with everything else(thanks for this brilliant piece of software!).
The date of the server and computer is correct.
Logs show nothing, just that coursera.org returns 301.

**Comment by aaronsinclair** [comment]
HI. Just working this out atm as well... here is some cursory information..

1. request is made in firefox for https://www.coursera.org/
2. pcap capture file is created using # tcpdump -nn host www.coursera.org -w coursera.pcap (Download from http://223.27.30.52/coursera.pcap.gz )
3. It appears that the TLS fails and then for some reason the request reverts to HTTP Site 
4. the HTTP site (http://www.coursera.org/) replies with a 301 redirect to the HTTPS (https://www.coursera.org/), thereby establishing the loop.


![capture](https://cloud.githubusercontent.com/assets/15620481/11266786/09699980-8f0c-11e5-9cfe-22a0ad1c68c8.PNG)

**Comment by fifthsegment** [comment]
So, it's now been almost 24 hours that I've been playing with e2g. SSLMITM works very well with most of the sites; I've experienced the redirect issue on some subreddits on reddit.com (happens very infrequently like 10% of the time) and is normally resolved by a refresh. 

Why would the TLS fail on coursera.org?

**Comment by philipianpearce** [comment]
Hi Abdullah, 

Could you post the related e2g access.log entries? 

Regards 

Philip 

----- Original Message -----

**Comment by fifthsegment** [comment]
Sure...I've just experienced this issue on a Facebook.com events link as well...check out the log entries.
-
[access.log.txt](https://github.com/e2guardian/e2guardian/files/38813/access.log.txt)

**Comment by aaronsinclair** [comment]
HI. 

Not sure if this helps..... 

301 REDIRECT from HTTP -> HTTPS
root@proxy:/tmp# wget http://www.coursera.org
--2015-11-19 22:54:34--  http://www.coursera.org/
Resolving www.coursera.org (www.coursera.org)... 54.192.134.179, 54.230.133.150, 54.192.132.74, ...
Connecting to www.coursera.org (www.coursera.org)|54.192.134.179|:80... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: https://www.coursera.org/ [following]
--2015-11-19 22:54:34--  https://www.coursera.org/
Connecting to www.coursera.org (www.coursera.org)|54.192.134.179|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 758647 (741K) [text/html]
Saving to: `index.html'

(debug log here: http://223.27.30.52/debug1.gz )
# e2guardian.debug > debug1
Socket 0 fd:6
bindSingle binding port8080
26859Start URL http://172.16.1.106/ssdp/device-desc.xmlis_ssl=0ismitm=0
26860Start URL http://172.16.1.110/upnphost/udhisapi.dll?content=uuid:0250da39-4524-4c57-aefd-402518d2ca51is_ssl=0ismitm=0
26861Start URL http://172.16.1.106/dd.xmlis_ssl=0ismitm=0
Got to final download manager so defaulting to always match.
26865Start URL https://www.coursera.orgis_ssl=1ismitm=0
1996190948:error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure:s23_clnt.c:749:
26865Start URL http://www.coursera.orgis_ssl=0ismitm=1
Got to final download manager so defaulting to always match.
26865Start URL http://www.coursera.orgis_ssl=0ismitm=1
Got to final download manager so defaulting to always match.
26865Start URL http://www.coursera.orgis_ssl=0ismitm=1
Got to final download manager so defaulting to always match.
26865Start URL http://www.coursera.orgis_ssl=0ismitm=1
Got to final download manager so defaulting to always match.
26865Start URL http://www.coursera.orgis_ssl=0ismitm=1
Got to final download manager so defaulting to always match.
26865Start URL http://www.coursera.orgis_ssl=0ismitm=1
Got to final download manager so defaulting to always match.
26865Start URL http://www.coursera.orgis_ssl=0ismitm=1
Got to final download manager so defaulting to always match.
26865Start URL http://www.coursera.orgis_ssl=0ismitm=1
Got to final download manager so defaulting to always match.
26865Start URL http://www.coursera.orgis_ssl=0ismitm=1
Got to final download manager so defaulting to always match.
26865Start URL http://www.coursera.orgis_ssl=0ismitm=1
Got to final download manager so defaulting to always match.
26865Start URL http://www.coursera.orgis_ssl=0ismitm=1
Got to final download manager so defaulting to always match.
26865Start URL http://www.coursera.orgis_ssl=0ismitm=1
Got to final download manager so defaulting to always match.
26865Start URL http://www.coursera.orgis_ssl=0ismitm=1
Got to final download manager so defaulting to always match.
26865Start URL http://www.coursera.orgis_ssl=0ismitm=1
Got to final download manager so defaulting to always match.
26865Start URL http://www.coursera.orgis_ssl=0ismitm=1
Got to final download manager so defaulting to always match.
26865Start URL http://www.coursera.orgis_ssl=0ismitm=1
Got to final download manager so defaulting to always match.
26865Start URL http://www.coursera.orgis_ssl=0ismitm=1
26866Start URL https://www.coursera.orgis_ssl=1ismitm=0
Got to final download manager so defaulting to always match.
26865Start URL http://www.coursera.orgis_ssl=0ismitm=1
26867Start URL https://www.coursera.orgis_ssl=1ismitm=0
Got to final download manager so defaulting to always match.
1996190948:error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure:s23_clnt.c:749:
26865Start URL http://www.coursera.orgis_ssl=0ismitm=1
Got to final download manager so defaulting to always match.
1996190948:error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure:s23_clnt.c:749:
26866Start URL http://www.coursera.orgis_ssl=0ismitm=1
Got to final download manager so defaulting to always match.
26865Start URL http://www.coursera.orgis_ssl=0ismitm=1
Got to final download manager so defaulting to always match.
Got to final download manager so defaulting to always match.
59123 -connection handler caught an exception: std::exception

# <snip> debug1

CheckHeader flags before normalisation: AP=0 PPC=0 1.1=0 connectionclose=1 CL=0
CheckHeader flags after normalisation: AP=0 WP=0
CheckHeader: Adding our own Proxy-Connection: Close
59125 -persistPeer: 0
59125 -Getting ssl certificate for client connection
Generating serial no for www.coursera.org
Serial no is 054B963CF6532198A7987E971CFF66A0
looking for cert /usr/local/etc/e2guardian/generatedcerts/05/4B/96/3CF6532198A7987E971CFF66A0
Certificate not found. Creating one
certificate create 0x1183028
59125 -Going SSL on the peer connection
59125 -Going SSL on connection to proxy
ssl_connect failed with error 5
59125 -Checking certificate
certificate file is /usr/local/etc/e2guardian/generatedcerts/05/4B/96/3CF6532198A7987E971CFF66A0
got first lock
59125 -Handling connections inside ssl tunnel
59125 -got peer connection
59125172.16.1.101
Host: www.coursera.org
Connection: keep-alive
Cache-Control: max-age=0
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8
Upgrade-Insecure-Requests: 1
User-Agent: Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/46.0.2490.86 Safari/537.36
Accept-Encoding: identity,gzip,deflate
Accept-Language: en-US,en;q=0.8
CheckHeader: HTTP/1.1 detected
CheckHeader flags before normalisation: AP=1 PPC=1 1.1=1 connectionclose=0 CL=0
CheckHeader flags after normalisation: AP=1 WP=0
from header url:http://www.coursera.org/
from header url:http://www.coursera.org/
setURL: header.front() changed from: GET http://www.coursera.org/ HTTP/1.0
 to: GET http://www.coursera.org/ HTTP/1.0
setURL: header[] line changed from: Host: www.coursera.org
 to Host: www.coursera.org
 firsttime =1ismitm =1 clientuser =172.16.1.101 group = 0
from header url:http://www.coursera.org
from header url:http://www.coursera.org

E2Guardian Log: 

1447927599.038          172.16.1.101    172.16.1.101    aaron_s3.secure.local   https://www.coursera.org        GET     301     183     -       -       -       110     -       0               0               StrictFilterMode        1
1447927599.109          172.16.1.101    172.16.1.101    aaron_s3.secure.local   https://www.coursera.org        GET     301     183     -       -       -       58      -       0               0               StrictFilterMode        1
1447927599.197          172.16.1.101    172.16.1.101    aaron_s3.secure.local   https://www.coursera.org        GET     301     183     -       -       -       59      -       0               0               StrictFilterMode        1
1447927599.294          172.16.1.101    172.16.1.101    aaron_s3.secure.local   https://www.coursera.org        GET     301     183     -       -       -       58      -       0               0               StrictFilterMode        1
1447927599.363          172.16.1.101    172.16.1.101    aaron_s3.secure.local   https://www.coursera.org        GET     301     183     -       -       -       57      -       0               0               StrictFilterMode        1
1447927599.437          172.16.1.101    172.16.1.101    aaron_s3.secure.local   https://www.coursera.org        GET     301     183     -       -       -       61      -       0               0               StrictFilterMode        1
1447927599.510          172.16.1.101    172.16.1.101    aaron_s3.secure.local   https://www.coursera.org        GET     301     183     -       -       -       63      -       0               0               StrictFilterMode        1
1447927599.615          172.16.1.101    172.16.1.101    aaron_s3.secure.local   https://www.coursera.org        GET     301     183     -       -       -       68      -       0               0               StrictFilterMode        1
1447927599.715          172.16.1.101    172.16.1.101    aaron_s3.secure.local   https://www.coursera.org        GET     301     183     -       -       -       63      -       0               0               StrictFilterMode        1
1447927599.785          172.16.1.101    172.16.1.101    aaron_s3.secure.local   https://www.coursera.org        GET     301     183     -       -       -       63      -       0               0               StrictFilterMode        1
1447927599.862          172.16.1.101    172.16.1.101    aaron_s3.secure.local   https://www.coursera.org        GET     301     183     -       -       -       63      -       0               0               StrictFilterMode        1
1447927599.943          172.16.1.101    172.16.1.101    aaron_s3.secure.local   https://www.coursera.org        GET     301     183     -       -       -       63      -       0               0               StrictFilterMode        1
1447927600.029          172.16.1.101    172.16.1.101    aaron_s3.secure.local   https://www.coursera.org        GET     301     183     -       -       -       64      -       0               0               StrictFilterMode        1
1447927600.107          172.16.1.101    172.16.1.101    aaron_s3.secure.local   https://www.coursera.org        GET     301     183     -       -       -       63      -       0               0               StrictFilterMode        1
1447927600.214          172.16.1.101    172.16.1.101    aaron_s3.secure.local   https://www.coursera.org        GET     301     183     -       -       -       64      -       0               0               StrictFilterMode        1
1447927600.314          172.16.1.101    172.16.1.101    aaron_s3.secure.local   https://www.coursera.org        GET     301     183     -       -       -       65      -       0               0               StrictFilterMode        1
1447927600.404          172.16.1.101    172.16.1.101    aaron_s3.secure.local   https://www.coursera.org        GET     301     183     -       -       -       65      -       0               0               StrictFilterMode        1
1447927600.538          172.16.1.101    172.16.1.101    aaron_s3.secure.local   https://www.coursera.org        GET     301     183     -       -       -       65      -       0               0               StrictFilterMode        1
1447927600.618          172.16.1.101    172.16.1.101    aaron_s3.secure.local   https://www.coursera.org        GET     301     183     -       -       -       63      -       0               0               StrictFilterMode        1
1447927600.693          172.16.1.101    172.16.1.101    aaron_s3.secure.local   https://www.coursera.org        GET     301     183     -       -       -       63      -       0               0               StrictFilterMode        1
1447927600.800          172.16.1.101    172.16.1.101    aaron_s3.secure.local   https://www.coursera.org        GET     301     183     -       -       -       63      -       0               0               StrictFilterMode        

SQUID LOG: 

1447927598.927    414 172.16.1.101 TCP_MISS/200 7 CONNECT www.coursera.org:443 - DIRECT/54.192.134.48 -
1447927599.030    100 172.16.1.101 TCP_MISS/301 660 GET http://www.coursera.org/ - DIRECT/54.192.134.48 text/html
1447927599.102     50 172.16.1.101 TCP_MISS/301 660 GET http://www.coursera.org/ - DIRECT/54.192.134.48 text/html
1447927599.190     51 172.16.1.101 TCP_MISS/301 660 GET http://www.coursera.org/ - DIRECT/54.192.134.48 text/html
1447927599.287     50 172.16.1.101 TCP_MISS/301 660 GET http://www.coursera.org/ - DIRECT/54.192.134.48 text/html
1447927599.361     55 172.16.1.101 TCP_MISS/301 660 GET http://www.coursera.org/ - DIRECT/54.192.134.48 text/html
1447927599.434     57 172.16.1.101 TCP_MISS/301 660 GET http://www.coursera.org/ - DIRECT/54.192.134.48 text/html
1447927599.500     52 172.16.1.101 TCP_MISS/301 660 GET http://www.coursera.org/ - DIRECT/54.192.134.48 text/html
1447927599.605     56 172.16.1.101 TCP_MISS/301 660 GET http://www.coursera.org/ - DIRECT/54.192.134.48 text/html
1447927599.705     51 172.16.1.101 TCP_MISS/301 660 GET http://www.coursera.org/ - DIRECT/54.192.134.48 text/html
1447927599.774     51 172.16.1.101 TCP_MISS/301 660 GET http://www.coursera.org/ - DIRECT/54.192.134.48 text/html
1447927599.852     51 172.16.1.101 TCP_MISS/301 660 GET http://www.coursera.org/ - DIRECT/54.192.134.48 text/html
1447927599.933     51 172.16.1.101 TCP_MISS/301 660 GET http://www.coursera.org/ - DIRECT/54.192.134.48 text/html
1447927600.019     52 172.16.1.101 TCP_MISS/301 660 GET http://www.coursera.org/ - DIRECT/54.192.134.48 text/html
1447927600.097     51 172.16.1.101 TCP_MISS/301 660 GET http://www.coursera.org/ - DIRECT/54.192.134.48 text/html
1447927600.204     52 172.16.1.101 TCP_MISS/301 660 GET http://www.coursera.org/ - DIRECT/54.192.134.48 text/html
1447927600.303     52 172.16.1.101 TCP_MISS/301 660 GET http://www.coursera.org/ - DIRECT/54.192.134.48 text/html
1447927600.394     53 172.16.1.101 TCP_MISS/301 660 GET http://www.coursera.org/ - DIRECT/54.192.134.48 text/html
1447927600.526     52 172.16.1.101 TCP_MISS/301 660 GET http://www.coursera.org/ - DIRECT/54.192.134.48 text/html
1447927600.608     51 172.16.1.101 TCP_MISS/301 660 GET http://www.coursera.org/ - DIRECT/54.192.134.48 text/html
1447927600.683     51 172.16.1.101 TCP_MISS/301 660 GET http://www.coursera.org/ - DIRECT/54.192.134.48 text/html
1447927600.790     51 172.16.1.101 TCP_MISS/301 660 GET http://www.coursera.org/ - DIRECT/54.192.134.48 text/html

**Comment by aaronsinclair** [comment]
Im assuming that the failure (Going SSL on connection to proxy) is where the request reverts to http..
debug log: 
59125 -Going SSL on connection to proxy
ssl_connect failed with error 5

**Comment by fifthsegment** [comment]
Could this be a SNI issue?

**Comment by philipianpearce** [comment]
Thanks Aaron for the debug trace - really helpful 

Yes, the looping problem is caused by the openssl library connect call failing. 

There are two issues:- 

1/ E2g is not returning an error to the user when the ssl_connect fails, but is carrying on and making the http request directly to the cache instead of via an ssl tunnel. Hence the looping. I will fix this and post to the develop branch next week. 

2/ openssl library connect call failing with error 5. This appears to be a poorly documented openssl error (no error text, just number) possibly due to the DH negotiation with the target server failing. 

I have now been able to replicate the problem by downgrading the openssl ca-bundle.crt to an older version. Re-installing the latest ca-bundle.crt fixes the problem. 

So try getting and installing the latest ca-bundle. 

Regards 
Philip

**Comment by aaronsinclair** [comment]
Two thumbs up! Thanks Philip- the two issues I have left on my system are these 302 and 301 redirect loops, and the SSL Cert Errors I've been getting. CA Bundle was already earmarked for update to resolve issue 2 - just the the order of tasks was arse about. 

Many thanks for your fine work- let me know if there is any way to repay the favor

**Comment by fifthsegment** [comment]
Does updating the ca-bundle resolve the redirect issue for you guys? because in my case it's not working. Maybe I'm not doing it correctly. Could someone post how to correctly update ca-bundle on ubuntu?

Here's what I've tried:
* Downloaded firefox's latest CA cert store file to /usr/local/share/ca-certificates/
* Downloaded http://www.cacert.org/certs/root.crt http://www.cacert.org/certs/class3.crt to /usr/local/share/ca-certificates/
* Ran the sudo update-ca-certificates command

But, I still face the redirect issue.

**Comment by aaronsinclair** [comment]
Hi. First I checked that I had the latest ca-certifictes package. 

.# sudo  apt-get update
.# sudo apt-get install ca-certificates
> ca-certificates is already the newest version.

So decieded to use curl.haxx.se/ca/cacert.pem ca bundle
.# cd /usr/share/ca-certificates/
.# wget http://curl.haxx.se/ca/cacert.pem
.# update-ca-certificates

**Comment by fifthsegment** [comment]
Updated certs but still the same issue. 
I even tried it on another machine with Debian 8, by first updating the ca-bundle then installing e2G. I'm still getting the redirect loop :(

PS: I'm using the e2g v3.2.0 via the deb file.

**Comment by aaronsinclair** [comment]
HI. Actually, this is NOT working for me either... I thought it was fixed through updating ca-cert bundle, but I still have the same issues. 

-- If I have SSL verification disabled, why is the resource fetching checking for SSL validity anyway?
-- if it is not the validity that is the issue, then what good will updating ca-bundle do?

https://www.coursera.org appears to be a SNI issue

> openssl s_client -CApath /etc/ssl/certs/ -connect www.coursera.org:443
```CONNECTED(00000003)
1995941072:error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure:s23_clnt.c:749:
---
no peer certificate available
---
No client certificate CA names sent
---
SSL handshake has read 7 bytes and written 298 bytes
---
New, (NONE), Cipher is (NONE)
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
---
```

>  openssl s_client -CApath /etc/ssl/certs/ -connect www.coursera.org:443 -servername  www.coursera.org
```
CONNECTED(00000003)
depth=3 C = US, O = "The Go Daddy Group, Inc.", OU = Go Daddy Class 2 Certification Authority
verify return:1
depth=2 C = US, ST = Arizona, L = Scottsdale, O = "GoDaddy.com, Inc.", CN = Go Daddy Root Certificate Authority - G2
verify return:1
depth=1 C = US, ST = Arizona, L = Scottsdale, O = "GoDaddy.com, Inc.", OU = http://certs.godaddy.com/repository/, CN = Go Daddy Secure Certificate Authority - G2
verify return:1
depth=0 OU = Domain Control Validated, CN = *.coursera.org
verify return:1
---
Certificate chain
 0 s:/OU=Domain Control Validated/CN=*.coursera.org
   i:/C=US/ST=Arizona/L=Scottsdale/O=GoDaddy.com, Inc./OU=http://certs.godaddy.com/repository//CN=Go Daddy Secure Certificate Authority - G2
 1 s:/C=US/ST=Arizona/L=Scottsdale/O=GoDaddy.com, Inc./OU=http://certs.godaddy.com/repository//CN=Go Daddy Secure Certificate Authority - G2
   i:/C=US/ST=Arizona/L=Scottsdale/O=GoDaddy.com, Inc./CN=Go Daddy Root Certificate Authority - G2
 2 s:/C=US/ST=Arizona/L=Scottsdale/O=GoDaddy.com, Inc./CN=Go Daddy Root Certificate Authority - G2
   i:/C=US/O=The Go Daddy Group, Inc./OU=Go Daddy Class 2 Certification Authority
---<snip>
 Start Time: 1448395554
    Timeout   : 300 (sec)
    Verify return code: 0 (ok)
---
closed
```

**Comment by fifthsegment** [comment]
Yeah, that's [what](https://github.com/e2guardian/e2guardian/issues/92#issuecomment-158015331) I thought too, because it occurs only on some very specific websites.

**Comment by aaronsinclair** [comment]
Here's another site that requires -servername addition.. 
openssl s_client -CApath /etc/ssl/certs/ -connect www.familyfirst.org.nz:443 -servername www.familyfirst.org.nz

**Comment by philipianpearce** [comment]
I have posted a fix so that SNI is sent when requesting certs from sites to the develop branch.
However only works with openssl v1.0.1e + as it is not supported in openssl 0.9x.

However there may also be other reasons for the openssl error 5 as this is so non-specific.

Changing the cert bundle did work for me for some of the sites, as it appears that for some sites info is required from the rootCA certs in order to complete the SSL connection.  So certs appear to be needed even when you have certificate checking turned off.   (The cert checking is done after the SSL connection has been made).

**Comment by aaronsinclair** [comment]
www.coursera.org and www.familyfirst.org.nz - the two domains that I assumed to be SNI are both working with your development branch fix.. Thank you!

Here is a different case to the previous ones: 

.# curl -I "https://cart.rackspace.com/fp/clear.png?org_id=k8vif92e&session_id=rackspace2af03b084140526983c58b64c54b3c356dc8a4dc3b5dcf026e4e86df0eae52f5&m=2"
```
HTTP/1.1 303 See Other
Date: Thu, 26 Nov 2015 03:57:00 GMT
Server: Apache
X-Frame-Options: SAMEORIGIN
Location: https://h.online-metrix.net/fp/clear.png?org_id=k8vif92e&session_id=rackspace2af03b084140526983c58b64c54b3c356dc8a4dc3b5dcf026e4e86df0eae52f5&m=2
Cache-Control: max-age=0
Expires: Thu, 26 Nov 2015 03:57:00 GMT
Vary: Accept-Encoding
Content-Type: text/html; charset=iso-8859-1
Set-Cookie: BIGip=!E31SnemmOT6qo7hAzI3dEckTHDH4Hj4OSMNDEA9Z/aQ/68Ox3C+TMloGCqJdCks/l9Rgt5M0ir6gO+U=; path=/
Set-Cookie: IS_UASrackuid=USc7c8341b86afba8719754ee91c522627;expires=Sun, 25-Nov-2035 23:57:01 GMT; path=/; domain=.rackspace.com
Set-Cookie: RackSID=20151125215701; path=/; domain=.rackspace.com
```

This results in an ongoing loop

**Comment by fifthsegment** [comment]
Hi guys, is content filtering disabled by default on the develop branch? I just pulled `develop` and did a `make install` via it only to find that content filtering doesn't seem to be working. 
URL blocking works fine though.

I then reverted back to the master branch; content blocking works fine on it.

**Comment by fredbcode** [comment]
No I'm just working on it, weightedphrasemode is now moved to e2guardianfx.conf 
At the end we will remove all the filtering rules from e2guardian.conf, this file well be dedicated to engine parameters

**Comment by fifthsegment** [comment]
Oh okay, thanks! So content filtering is now working by adding weightedphrasemode, but sslmitm is not (I'm using both e2guardian.conf and e2guardianfx.conf files from my previous e2g install from the master in which ssl filtering was working).

Do I need to add any thing else to my e2guardianfx.conf file?

**Comment by fredbcode** [comment]
You mean content filtering with SSLMITM or just SSLMITM ? 
Please share your config file

**Comment by fifthsegment** [comment]
Sure, my files are attached. I have SSLMITM turned on, and I want https traffic to be filtered (content filtering with SSLMITM).
[e2guardian.conf](https://gist.github.com/fifthsegment/f733534bbbe06c1eec40)
[e2guardianf1.conf](https://gist.github.com/fifthsegment/042a2732493969e66533)

Thanks

**Comment by fredbcode** [comment]
Sorry I'm not clear, do you think SSLMITM is broken OR SSLMITM with content filtering

**Comment by aaronsinclair** [comment]
That's a shame. progress had been made on the SSL redirect loop, but it appears to have diverged off topic into a general support request. For what it's worth MITM and SSL filtering are all working on the current develop branch. 

This redirect loop is the main issue that I face. Please let me know I f there is anything at all I can do to assist.

**Comment by fifthsegment** [comment]
@aaronsinclair You're right, is there some place else or a discussion forum where I can take this to?
@fredbcode The issue I'm facing is that my https traffic isn't getting filtered at all by content filtering.

**Comment by fredbcode** [comment]
Yes https://groups.google.com/forum/#!forum/e2guardian

**Comment by fifthsegment** [comment]
My small issue was resolved. I'm now running e2g from the develop branch, my openSSL version is 1.0.1f 6 Jan 2014.

The domain [www.familyfirst.org.nz](http://www.familyfirst.org.nz) opens fine but coursera still doesn't.

**Comment by philipianpearce** [comment]
This appears to be due to cert check failure in openssl for xxxxx.cloudfront.net - see http://stackoverflow.com/questions/27804710/python-urllib2-ssl-error for more info. 

Looks as if need to upgrade to openssl 1.0.2 for a proper fix.

Meantime you could add cloudfront.net to your nocheckcertsitelist as a work round.

**Comment by fifthsegment** [comment]
Can confirm Coursera.org opening fine after adding cloudfront to nocheckcertsitelist. Thanks!

**Comment by nminica** [comment]
Is this issue fixed in e2guardian 3.4.0.3?
